File: Web/Engine/Primitive/AxBoundingVolume.js
/**
 * Represents the bounding volume of a volumetric object
 * @constructor
 */
function AxBoundingVolume()
{
    this.center = new AxVector3();
    this.box = new AxVector3();
    
    this.sphereRadius = 0.0;
}

/**
 * Returns whether the bounding volume is inside the view frustum of the given camera
 * @param {AxMatrix} viewFrustum The world-view-projection transformation to check against
 * @return {Boolean} True if the bounding volume is in the given transormation's view frustum
 */
AxBoundingVolume.prototype.InsideViewFrustum = function(viewFrustum)
{
    return AxMaths.SphereViewFrustumIntersection(viewFrustum, this.center, this.sphereRadius);
};

File: Engine/Axis/Source/Engine/Primitive/AxBoundingVolume.cpp
//---------------------------------------------------------------------------------------------
//	Axis 3D Engine Copyright (C) 2017 Palitri
//
//	This software is released under the MIT License http://palitri.com/axis/license/
//
//	For help and documentation, visit http://palitri.com
//---------------------------------------------------------------------------------------------

#include "AxBoundingVolume.h"

#include "AxMatrix.h"
#include "AxVector4.h"

#include "..\Utilities\AxMaths.h"

AxBoundingVolume::AxBoundingVolume(void)
{
}


AxBoundingVolume::~AxBoundingVolume(void)
{
}

bool AxBoundingVolume::InsideViewFrustum(AxMatrix &viewFrustum)
{
	return AxMaths::SphereViewFrustumIntersection(viewFrustum, this->center, this->sphereRadius);
}
